frontend/src/components/admin/SearchUsersBar.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import {
+  Input,
+  InputGroup,
+  InputRightElement,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+} from "@chakra-ui/react";
+import { SearchIcon } from "@chakra-ui/icons";
+import { SearchUsersBarProps } from "../../types/AdminDashboardTypes";
+
+const SearchUsersBar = ({
+  users,
+  onUserSelect,
+}: SearchUsersBarProps): React.ReactElement => {
+  const [searchEmail, setSearchEmail] = useState<string>("");
+  const [showNewResult, setShowNewResult] = useState<boolean>(false);
+
+  const handleOnSelectUser = () => {
+    onUserSelect(searchEmail);
+    setSearchEmail("");
+  };
+
+  return (
+    <Menu
+      isOpen={searchEmail !== "" && showNewResult}
+      offset={[8, 20]}
+      onClose={() => setShowNewResult(false)}
+    >
+      <MenuButton tabIndex={-1} />
+      <InputGroup w="500px">
+        <Input
+          value={searchEmail}
+          placeholder="Search users by email"
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+            setSearchEmail(event.target.value);
+            setShowNewResult(true);
+          }}
+          onFocus={() => setShowNewResult(true)}
+          onKeyPress={(event: React.KeyboardEvent<HTMLInputElement>) => {
+            if (event.key === "Enter" && searchEmail in users) {
+              handleOnSelectUser();
+            }
```

Comment:
100% recommend testing this out yourself in staging/locally to see this interaction (both through clicks and using keyboard)